pkg/controller/operators/catalog/step.go

```diff
@@ -139,7 +139,13 @@ func (b *builder) NewCRDV1Step(client apiextensionsv1client.ApiextensionsV1Inter
 					}
 				}
 
-				// TODO ensure stored version compatibility
+				// check to see if stored versions changed and whether the upgrade could cause potential data loss
+				err = crdlib.SafeStorageVersionUpgrade(currentCRD, crd)
+				if err != nil {
+					return v1alpha1.StepStatusUnknown, errorwrap.Wrapf(err, "potential for data loss updating: %s", step.Resource.Name)
```

Comment:
can we make this a little more descriptive? i.e. "potential for data loss: new CRD removes <version> that is listed a as a stored version on the existing CRD."

maybe even "consult docs for CRD storage version for how to safely remove." etc

Reply:
so this wraps the error coming from the function which would add `"could not find existing stored version %s in the spec of the new CRD", name`. I can log the error out and take a look to make it as descriptive as possible 👍